kitty_tests/history.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+# vim:fileencoding=utf-8
+# License: GPL v3
+
+from . import BaseTest
+from kitty.config import defaults
+from kitty.fast_data_types import set_options
+
+class TestHistory(BaseTest):
+
+    def test_history_short(self):
+        defaults.scrollback_pager_history_size = 10
+        set_options(defaults)
+        s = self.create_screen()
+
+        for i in range(1, 1000*1000):
+            s.draw('aaa')
+            s.linefeed()
+            s.carriage_return()
+
+    def test_history_long(self):
+        defaults.scrollback_pager_history_size = 10
```

Comment:
Change that to 0 to compare with/without; 0 with PR isn't too far from without PR (need better measurements).
The test still shows a bigger difference than what I'm observing in a real kitty instance with a huge cat, though; I haven't gotten to that yet.